File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/repository/AccountRepository.java
package com.lattis.ellipse.domain.repository;

import com.lattis.ellipse.domain.model.Account;

import java.util.List;

import io.reactivex.Observable;

public interface AccountRepository {

    Observable<List<Account>> getAccounts();

    Observable<Account> getAccount();

    Observable<String> getToken();

    Observable<Void> setToken(String tokenType, String tokenValue);

    Observable<String> getToken(String tokenType);

    Observable<Void> invalidateToken();

    Observable<Cloneable> addAccount();

    Observable<Account> addAccountExplicitly(Account account);

    Observable<Boolean> signOut();

    Observable<Void> invalidateToken(String tokenType);

    Observable<List<Account>> getAccounts(String accountType);

    Observable<Account> getAccount(String accountType);

    Observable<String> getToken(Account account, String tokenType);

    Observable<Void> invalidateToken(String accountType, String token);

}

File: velolabs_repos/operator-android/domain/src/main/java/io/lattis/domain/repository/AccountRepository.kt
package io.lattis.domain.repository

import io.lattis.domain.models.Account
import io.reactivex.Observable

interface AccountRepository {

    fun getAccounts(): Observable<List<Account>>

    fun getAccount(): Observable<Account>

    fun getToken(): Observable<String>

    fun setToken(tokenType: String, tokenValue: String): Observable<Void>

    fun getToken(tokenType: String): Observable<String>

    fun invalidateToken(): Observable<Void>

    fun addAccount(): Observable<Cloneable>

    fun addAccountExplicitly(account: Account?): Observable<Account>

    fun signOut(): Observable<Boolean>

    fun invalidateToken(tokenType: String): Observable<Void>

    fun getAccounts(accountType: String): Observable<List<Account>>

    fun getAccount(accountType: String): Observable<Account>

    fun getToken(account: Account, tokenType: String): Observable<String>

    fun invalidateToken(accountType: String, token: String): Observable<Void>

}
